addon/webextension/makeUuid.js

```diff
@@ -1,10 +1,17 @@
-window.makeUuid = (function () {
+"use strict";
 
-  // From http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript
+this.makeUuid = (function() {
+
+  // generates a v4 UUID
   return function makeUuid() { // eslint-disable-line no-unused-vars
+    // get sixteen unsigned 8 bit random values
+    var randomValues = window
+      .crypto
+      .getRandomValues(new Uint8Array(36));
+
     return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
-      var r = Math.random()*16 | 0,
-        v = c === 'x' ? r : (r & 0x3 | 0x8);
+      var i = Array.prototype.slice.call(arguments).slice(-2)[0]; // grab the `offset` parameter
```

Comment:
If you really want to do the Array.prototype.slice thing, this should be written as `Array.prototype.slice.call(arguments, -2)`. But I'm not sure why you aren't just explicitly specifying this in the arguments list. Failing that, you can use a rest arg `function(c, ...args)` and then `args.slice(-2)[0]`